Emby.Server.Implementations/Syncplay/SyncplayController.cs

```diff
@@ -228,190 +271,271 @@ public void SessionJoin(SessionInfo session, JoinGroupRequest request)
+
+                        _group.LastActivity = currentTime.AddMilliseconds(
+                            delay
+                        );
+
+                        var command = NewSyncplayCommand(SendCommandType.Play);
+                        SendCommand(session, BroadcastType.AllGroup, command, CancellationToken.None);
+                    }
                 }
             }
-            else if (request.Type.Equals(PlaybackRequestType.UpdatePing))
+            else
             {
-                _group.UpdatePing(session, request.Ping ??= _group.DefaulPing);
+                // Group was not waiting, make sure client has latest state
+                var command = NewSyncplayCommand(SendCommandType.Play);
+                SendCommand(session, BroadcastType.CurrentSession, command, CancellationToken.None);
             }
         }
 
+        /// <summary>
+        /// Sanitizes the PositionTicks, considers the current playing item when available.
+        /// </summary>
+        /// <param name="positionTicks">The PositionTicks.</param>
+        /// <value>The sanitized PositionTicks.</value>
+        private long SanitizePositionTicks(long? positionTicks)
+        {
+            var ticks = positionTicks ??= 0;
+            ticks = ticks >= 0 ? ticks : 0;
+            if (_group.PlayingItem.RunTimeTicks != null)
+            {
+                var runTimeTicks = _group.PlayingItem.RunTimeTicks ??= 0;
+                ticks = ticks > runTimeTicks ? runTimeTicks : ticks;
+            }
+            return ticks;
+        }
+
+        /// <summary>
+        /// Updates ping of a session.
+        /// </summary>
+        /// <param name="session">The session.</param>
+        /// <param name="request">The update.</param>
+        private void HandlePingUpdateRequest(SessionInfo session, PlaybackRequest request)
+        {
+            // Collected pings are used to account for network latency when unpausing playback
+            var ping = request.Ping ??= _group.DefaulPing;
+            _group.UpdatePing(session, ping);
+        }
+
         /// <inheritdoc />
         public GroupInfoView GetInfo()
         {
-            var info = new GroupInfoView();
-            info.GroupId = GetGroupId().ToString();
-            info.PlayingItemName = _group.PlayingItem.Name;
-            info.PlayingItemId = _group.PlayingItem.Id.ToString();
-            info.PositionTicks = _group.PositionTicks;
-            info.Partecipants = _group.Partecipants.Values.Select(session => session.Session.UserName).ToArray();
-            return info;
+            return new GroupInfoView()
+            {
+                GroupId = GetGroupId().ToString(),
+                PlayingItemName = _group.PlayingItem.Name,
+                PlayingItemId = _group.PlayingItem.Id.ToString(),
+                PositionTicks = _group.PositionTicks,
+                Participants = _group.Participants.Values.Select(session => session.Session.UserName).ToArray()
```

Comment:
so we identify sessions by their users' names, not by some session ids? why?

Reply:
This is similar to the "buffering message", it's meant (for now) to be seen at a user level, specifically when joining a group, the user sees this list of participants.

I just noticed that there might be duplicates of the same user that could be removed to keep the list cleaner.

In the future, more detailed info could be sent to the clients.